misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/EvaluationUtils.java

```diff
@@ -105,7 +109,31 @@ public static BExpressionValue make(SuspendedContext context, double val) {
         return new BExpressionValue(context, context.getAttachedVm().mirrorOf(val));
     }
 
-    public static BExpressionValue make(SuspendedContext context, String val) {
-        return new BExpressionValue(context, context.getAttachedVm().mirrorOf(val));
+    public static BExpressionValue make(SuspendedContext context, String val) throws EvaluationException {
+        val = removeRedundantQuotes(val);
+        Value bStringVal = getAsBString(context, val);
+        return new BExpressionValue(context, bStringVal);
+    }
+
+    /**
+     * Converts the user given string literal into an {@link org.ballerinalang.jvm.api.values.BString} instance.
+     *
+     * @param context suspended debug context
+     * @param val     string value
+     * @return {@link org.ballerinalang.jvm.api.values.BString} instance
+     */
+    private static Value getAsBString(SuspendedContext context, String val) throws EvaluationException {
+        List<ReferenceType> cls = context.getAttachedVm().classesByName(B_STRING_UTILS_CLASS);
+        if (cls.isEmpty()) {
+            cls = Collections.singletonList(loadClass(context, B_STRING_UTILS_CLASS, "fromString"));
```

Comment:
We can use the FROM_STRING_METHOD constant here too.

Reply:
Yeah seems I've missed it somehow. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/26050/commits/32fbe2ec4c6d0b5ef068996df00abd3b4c245a61. Thanks!